src/map.cpp

```diff
@@ -2646,21 +2643,21 @@ std::pair<bool, bool> map::bash( const tripoint &p, const int str,
         bool smash_ter = false;
         map_bash_info *bash = NULL;
 
-        if ( has_furn(x, y) && furn_at(x, y).bash.str_max != -1 ) {
-            bash = &(furn_at(x,y).bash);
+        if ( has_furn(p) && furn_at(p).bash.str_max != -1 ) {
+            bash = &(furn_at(p).bash);
             smash_furn = true;
-        } else if ( ter_at(x, y).bash.str_max != -1 ) {
-            bash = &(ter_at(x,y).bash);
+        } else if ( ter_at(p).bash.str_max != -1 ) {
+            bash = &(ter_at(p).bash);
             smash_ter = true;
         }
         // TODO: what if silent is true?
-        if (has_flag("ALARMED", x, y) && !g->event_queued(EVENT_WANTED)) {
-            sounds::sound(x, y, 40, _("an alarm go off!"));
-            // if the player is nearby blame him/her
-            if( rl_dist( g->u.posx(), g->u.posy(), x, y ) <= 3 ) {
+        if (has_flag("ALARMED", p) && !g->event_queued(EVENT_WANTED)) {
+            sounds::sound(p, 40, _("an alarm go off!"));
+            // Blame nearby player
+            if( rl_dist( g->u.pos(), p ) <= 3 ) {
                 g->u.add_memorial_log(pgettext("memorial_male", "Set off an alarm."),
                                       pgettext("memorial_female", "Set off an alarm."));
-                const point abs = overmapbuffer::ms_to_sm_copy( getabs( x, y ) );
+                const point abs = overmapbuffer::ms_to_sm_copy( getabs( p.x, p.y ) );
```

Comment:
Both functions (`getabs` and `ms_to_sm_copy`) should work with tripoints, do they not?